### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by sindkor**
- [x] Deploys AVD-resources
 - [x] Deploys VMs
 - [x] Deploys extensions for AAD- and hostpool-join
 - [x] Deploys role assignments for users and admins
 - [x] Testing with various configurations
   - Personal and pooled
   - Scale up and down
 - [x] Examples in readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,75 @@
 # azurerm-azure-virtual-desktop
-Terraform module for deploying Azure Virtual Desktop.
+Terraform module for deploying Azure Virtual Desktop. Deploys a signle personal or shared host pool.
```
Comment (nikolaifa) [inline README.md:2]: Spelling error: `single` not `single`

File: variables.tf [hunk 2]
```diff
@@ -7,3 +7,125 @@ variable "resource_group_name" {
...
+
+variable "local_admin_username" {
+  type        = string
+  description = "The username of the local admin account on the hosts."
+  default     = "azureuser"
+}
+
+variable "local_admin_password" {
+  type        = string
+  description = "The password of the local admin account on the hosts. Defaults to a randomly generated password. This will be saved in state."
```
Comment (nikolaifa) [inline variables.tf:56]: Mention that it defaults to randomly generated **unique** passwords for each host

File: local.tf [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  host_pool_name                 = var.host_pool_name_override == "" ? "${var.system_name}-hostpool" : var.host_pool_name_override
+  workspace_name                 = var.workspace_name_override == "" ? "${var.system_name}-workspace" : var.workspace_name_override
+  desktop_application_group_name = var.desktop_application_group_name_override == "" ? "${var.system_name}-appgroup" : var.desktop_application_group_name_override
+
+  host_pool_load_balancer_type = var.host_pool_type == "Personal" ? "Persistent" : var.host_pool_load_balancer_type
+
+  custom_rdp_properties = join(";",
+    flatten([
+      var.custom_rdp_properties,
+      var.aad_joined_allow_access_from_nonjoined ? ["targetisaadjoined:i:1"] : [""]
```
Comment (nikolaifa) [inline local.tf:11]: I suggest adding booleans for managing other default RDP properties, e.g. disk or audio redirects.
https://docs.microsoft.com/en-us/azure/virtual-desktop/customize-rdp-properties#default-rdp-file-properties

This is probably out of scope for this PR.

**Comment by nikolaifa** [comment]
LGTM! ☺️ Have you verified the configuration with `terraform plan`?

File: variables.tf [hunk 4]
```diff
@@ -7,3 +7,125 @@ variable "resource_group_name" {
...
+}
+
+variable "host_location" {
+  type        = string
+  description = "The location to which the hosts (VMs) will be deployed."
+}
+
+variable "vm_size" {
+  type        = string
+  description = "The size of the hosts. E.g. `Standard_D2s_v3`."
```
[the remark this answers is not in the document]
Comment (sindkor) [reply variables.tf]: Good idea :)

File: local.tf (hunk 3 above)
Comment (sindkor) [reply local.tf:11]: Good idea. I'm not too happy about the readability of the code around how the custom RDP properties turned out. In my opinion there is too much going on in the code you referenced. In addition there is some logic in the host-pool resource itself.
I raised as suggested #6.

**Review by nikolaifa — APPROVED** [review]
LGTM

**Pull request merged**